Tests/scripts/hook_validations/pack_unique_files.py

```diff
@@ -0,0 +1,148 @@
+"""
+This module is designed to validate the existence and structure of content pack essential files in content.
+"""
+import os
+import io
+import re
+import json
+
+from Tests.test_utils import pack_name_to_path
+from Tests.scripts.constants import PACKS_WHITELIST_FILE_NAME, PACKS_PACK_IGNORE_FILE_NAME, PACKS_PACK_META_FILE_NAME, \
+    PACKS_README_FILE_NAME
+
+
+class PackUniqueFilesValidator():
+    """PackUniqueFilesValidator is designed to validate the correctness of pack's files structure.
+    Existence and validity of this files is essential."""
+
+    def __init__(self, pack):
+        self.pack = pack
+        self.pack_path = pack_name_to_path(self.pack)
+        self.secrets_file = PACKS_WHITELIST_FILE_NAME
+        self.pack_ignore_file = PACKS_PACK_IGNORE_FILE_NAME
+        self.pack_meta_file = PACKS_PACK_META_FILE_NAME
+        self.readme_file = PACKS_README_FILE_NAME
+        self._errors = []
+
+    # error handling
+    def _add_error(self, error):
+        """Adds error entry to a list under pack's name"""
+        if error:
+            self._errors.append(error)
+
+    def get_errors(self, raw=False):
+        """Get the dict version or string version for print"""
+        errors = ''
+        if raw:
+            errors = self._errors
+        elif self._errors:
+            errors = '@@@Issues with unique files in pack: {}\n  {}'.format(self.pack, '\n  '.join(self._errors))
+
+        return errors
+
+    # file utils
+    def _get_pack_file_path(self, file_name=''):
+        """Returns the full file path to pack's file"""
+        return os.path.join(self.pack_path, file_name)
+
+    def _is_pack_file_exists(self, file):
+        """Check if .secrets-ignore exists"""
+        if not os.path.isfile(self._get_pack_file_path(file)):
+            self._add_error('"{}" file does not exist, create one in the root of the pack'.format(file))
+            return False
+
+        return True
+
+    def _open_and_read_file_content(self, file_name):
```

Comment:
redundant, you can't read without open
```suggestion
    def _read_file_content(self, file_name):
```